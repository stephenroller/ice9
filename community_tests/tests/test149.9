# test recursive function call								succeed

proc foo(a : int) : int

	write a;
	if a > 10 ->
		foo := 0;
		return;
	fi
	
	foo := foo(a-1)+2;
end

write foo(20);