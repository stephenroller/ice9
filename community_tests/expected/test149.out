Loading file: generated/test149.tm
20 
0 
Number of instructions executed = 58
